### PR TITLE
refactor: Fluent Connection

### PR DESCRIPTION
**Pull request by prmukherj**
The FluentConnection class has been refactored. Please put in more suggestions if there as part of this PR.

### CODE REVIEW

**Review by mkundu1 — APPROVED** [review]

**Review by hpohekar — APPROVED** [review]

**Review by seanpearsonuk — APPROVED** [review]

File: src/ansys/fluent/core/fluent_connection.py [hunk 1]
```diff
@@ -201,6 +201,84 @@ def list_values(self) -> dict:
...
+class _ConnectionInterface:
+    def __init__(self, create_grpc_service, error_state):
+        self._scheme_eval_service = create_grpc_service(SchemeEvalService, error_state)
+        self.scheme_eval = service_creator("scheme_eval").create(
+            self._scheme_eval_service
+        )
+
+    @property
+    def product_build_info(self) -> str:
+        """Get Fluent build info."""
```
Comment (PipKat) [inline src/ansys/fluent/core/fluent_connection.py:237]:
```suggestion
        """Get Fluent build information."""
```
We should write out words we commonly abbreviate. If this is a property, the verb "Get" should be deleted. Properties are described as nouns.

File: src/ansys/fluent/core/fluent_connection.py [hunk 2]
```diff
@@ -201,6 +201,84 @@ def list_values(self) -> dict:
...
+            logger.debug("Obtaining Cortex connection properties...")
+            fluent_host_pid = self.scheme_eval.scheme_eval("(cx-client-id)")
+            cortex_host = self.scheme_eval.scheme_eval("(cx-cortex-host)")
+            cortex_pid = self.scheme_eval.scheme_eval("(cx-cortex-id)")
+            cortex_pwd = self.scheme_eval.scheme_eval("(cortex-pwd)")
+            logger.debug("Cortex connection properties successfully obtained.")
+        except _InactiveRpcError:
+            logger.warning(
+                "Fluent Cortex properties unobtainable, force exit and other"
+                "methods are not going to work properly, proceeding..."
```
Comment (PipKat) [inline src/ansys/fluent/core/fluent_connection.py:259]:
```suggestion
                "Fluent Cortex properties unobtainable. 'force exit()' and other"
                "methods are not going to work properly. Proceeding..."
```
These should be short sentences and not run-on sentences. What does the concluding "Proceeding" mean? Proceeding with what?

File: src/ansys/fluent/core/fluent_connection.py [hunk 3]
```diff
@@ -252,14 +329,12 @@ def __init__(
...
-        start_transcript : bool, optional
-            The Fluent transcript is started in the client only when
-            start_transcript is True. It can be started and stopped
-            subsequently via method calls on the Session object.
         remote_instance : ansys.platform.instancemanagement.Instance
             The corresponding remote instance when Fluent is launched through
             PyPIM. This instance will be deleted when calling
             ``Session.exit()``.
+        slurm_job_id: bool, optional
+            The job id of a Fluent session running within a Slurm environment.
```
Comment (PipKat) [inline src/ansys/fluent/core/fluent_connection.py:337]:
```suggestion
            Job ID of a Fluent session running within a Slurm environment.
```

File: src/ansys/fluent/core/fluent_connection.py [hunk 4]
```diff
@@ -252,14 +329,12 @@ def __init__(
...
         inside_container: bool, optional
             Whether the Fluent session that is being connected to
             is running inside a docker container.
```
Comment (PipKat) [inline src/ansys/fluent/core/fluent_connection.py:340]:
```suggestion
            is running inside a Docker container.
```

File: src/ansys/fluent/core/session.py [hunk 5]
```diff
@@ -93,13 +93,24 @@ def __init__(
...
         """BaseSession.
 
-        Args:
-            fluent_connection (:ref:`ref_fluent_connection`): Encapsulates a Fluent connection.
-            file_transfer_service: Supports file upload and download.
+        Parameters
+        ----------
+        fluent_connection (:ref:`ref_fluent_connection`):
+            Encapsulates a Fluent connection.
+        file_transfer_service:
```
Comment (PipKat) [inline src/ansys/fluent/core/session.py:105]:
```suggestion
        file_transfer_service :
```
Data type?

File: src/ansys/fluent/core/session.py [hunk 6]
```diff
@@ -93,13 +93,24 @@ def __init__(
...
+            Supports file upload and download.
+        start_transcript : bool, optional
+            The Fluent transcript is started in the client only when
+            start_transcript is True. It can be started and stopped
+            subsequently via method calls on the Session object.
```
Comment (PipKat) [inline src/ansys/fluent/core/session.py:110]:
```suggestion
            Whether to start the Fluent transcript in the client. The default
            is ``True``, in which case the Fluent transcript can be subsequently
            started and stopped using method calls on the ``Session`` object.
```

File: src/ansys/fluent/core/session.py [hunk 7]
```diff
@@ -264,9 +277,13 @@ def _create_from_server_info_file(
             Path to server-info file written out by Fluent server
         file_transfer_service : Optional
             Support file upload and download.
+        start_transcript : bool, optional
```
Comment (PipKat) [inline src/ansys/fluent/core/session.py:280]: Same suggestions as earlier.

File: src/ansys/fluent/core/fluent_connection.py (hunk 2 above)
Comment (prmukherj) [reply src/ansys/fluent/core/fluent_connection.py:259]: Thank you @PipKat. We'll get rid of it.

File: src/ansys/fluent/core/session.py (hunk 7 above)
Comment (prmukherj) [reply src/ansys/fluent/core/session.py:281]: Done, Thank you.

**Pull request merged**